src/parser/shared/metrics/apl/conditions/lastSpellCast.tsx

```diff
@@ -0,0 +1,36 @@
+import type Spell from 'common/SPELLS/Spell';
+import { SpellLink } from 'interface';
+import { EventType } from 'parser/core/Events';
+
+import { Condition } from '../index';
+
+export function lastSpellCast(spell: Spell): Condition<boolean> {
+  let sourceID: number;
```

Comment:
you know, i should have thought of (ab)using closures to track state instead of having this extra state parameter. On the one hand, explicit vs implicit + testability. on the other, one fewer parameter 🤔